core/src/main/scala/org/apache/spark/rpc/RpcTimeout.scala

```diff
@@ -125,9 +125,9 @@ private[spark] object RpcTimeout {
     var foundProp: Option[(String, String)] = None
     while (itr.hasNext && foundProp.isEmpty) {
       val propKey = itr.next()
-      conf.getOption(propKey).foreach { prop => foundProp = Some(propKey, prop) }
+      conf.getOption(propKey).foreach { prop => foundProp = Some((propKey, prop)) }
```

Comment:
Lots of the changes are of this form. Whereas scalac will still accept two args to a method that accepts one Tuple2 arg, it's a warning in Scala 2.11 in some cases. I actually also enabled all warnings of this form.